store/sqlstore/channel_store.go

```diff
@@ -769,53 +769,50 @@ func (s SqlChannelStore) get(id string, master bool, allowFromCache bool) (*mode
 }
 
 // Delete records the given deleted timestamp to the channel in question.
-func (s SqlChannelStore) Delete(channelId string, time int64) store.StoreChannel {
+func (s SqlChannelStore) Delete(channelId string, time int64) *model.AppError {
 	return s.SetDeleteAt(channelId, time, time)
 }
 
 // Restore reverts a previous deleted timestamp from the channel in question.
-func (s SqlChannelStore) Restore(channelId string, time int64) store.StoreChannel {
+func (s SqlChannelStore) Restore(channelId string, time int64) *model.AppError {
 	return s.SetDeleteAt(channelId, 0, time)
 }
 
 // SetDeleteAt records the given deleted and updated timestamp to the channel in question.
-func (s SqlChannelStore) SetDeleteAt(channelId string, deleteAt, updateAt int64) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		defer s.InvalidateChannel(channelId)
+func (s SqlChannelStore) SetDeleteAt(channelId string, deleteAt, updateAt int64) *model.AppError {
+	defer s.InvalidateChannel(channelId)
```

Comment:
@jespino I realize this is what we previously had, but wouldn't this run even with an aborted transaction? Would that invalidate a channel without deleting it?

Reply:
This invalidate the channel from the Cache, the side effect of this is removing a cache that wasn't needed to be removed. Not a big deal. Is super rare that the transaction fails here, and the effect is just that, the cache for this channel object get cleaned.

Reply:
Ok, great, thanks for clarifying!